### ISSUE TITLE
Supporting the encryption of vino server

### ISSUE THREAD

**Issue by mayanksuman** [TigerVNC/tigervnc#307]
TigerVNC client do not support the default encryption offered by vino-server on linux (Launchpad Bug [#1281250](https://bugs.launchpad.net/ubuntu/+source/vino/+bug/1281250)). One common workaround is to disable encryption in vino-server , but this leave the VNC connection unprotected.

So, support of default encryption offered by vino-server is much needed enhancement for TigerVNC client users, who use it to connect to their linux box.

**Comment by rfjakob** [comment]
Just posted a $20 bounty. [![Bountysource](https://api.bountysource.com/badge/issue?issue_id=34074059)](https://www.bountysource.com/issues/34074059-supporting-the-encryption-of-vino-server-auth-18?utm_source=34074059&utm_medium=shield&utm_campaign=ISSUE_BADGE)

**Comment by dcommander** [comment]
The TurboVNC Viewer now supports this feature in our latest stable build (http://www.turbovnc.org/DeveloperInfo/PreReleases under "master branch (2.1.x stable)").  Our viewer is a hybrid that uses Java (based originally on Brian's TigerVNC Java viewer code, but with many enhancements) for the GUI and C for performing JPEG decoding and other tasks that can't be performed in Java (such as twiddling low-level X11 stuff.)  It should be roughly as fast as the TigerVNC Viewer on most platforms.

It would be straightforward to port the feature from our viewer into TigerVNC's viewers, since the code structure is similar, but unfortunately $20 is not enough to cover my labor in doing that.

**Comment by a359566926** [comment]
> The TurboVNC Viewer now supports this feature in our latest stable build (http://www.turbovnc.org/DeveloperInfo/PreReleases under "master branch (2.1.x stable)"). Our viewer is a hybrid that uses Java (based originally on Brian's TigerVNC Java viewer code, but with many enhancements) for the GUI and C for performing JPEG decoding and other tasks that can't be performed in Java (such as twiddling low-level X11 stuff.) It should be roughly as fast as the TigerVNC Viewer on most platforms.
> 
> It would be straightforward to port the feature from our viewer into TigerVNC's viewers, since the code structure is similar, but unfortunately $20 is not enough to cover my labor in doing that.

Did you mean TurboVNC Viewer Viewer support vino server since 2.1.x? I use the newest windows relase(TurboVNC-2.2.5-x86.exe), but still got the "Server did not offer supported security type!" error.

**Comment by dcommander** [comment]
Only our Java viewer supports encryption, not the Windows native viewer.

**Comment by a359566926** [comment]
Thanks, I can connect to my vino server through the Java viewer now.
By the way, besides installing the Windows version and getting the Java viewer from the "java" folder, can I directly get the Java viewer(If so, which one should i download from the download page)?

**Comment by dcommander** [comment]
No, just install the Windows package and access "Java TurboVNC Viewer" from the Start Menu.
